SimCalorimetry/CastorSim/plugins/CastorDigiProducer.cc

```diff
@@ -61,10 +61,11 @@ class CastorDigiProducer : public DigiAccumulatorMixMod {
   /// exist in the geometry
   void checkGeometry(const edm::EventSetup &eventSetup);
 
-  edm::InputTag theHitsProducerTag;
   const edm::ESGetToken<CastorDbService, CastorDbRecord> theConditionsToken;
   const edm::ESGetToken<CaloGeometry, CaloGeometryRecord> theGeometryToken;
   edm::ESWatcher<CaloGeometryRecord> theGeometryWatcher;
+  const edm::InputTag theHitsProducerTag;
```

Comment:
This `theHitsProducerTag` can also get removed.
Before doing that please notice that one `getByLabel` was left in https://github.com/cms-sw/cmssw/blob/ea3ff2016b1d90a8e3dba032325260bd44609bd1/SimCalorimetry/CastorSim/plugins/CastorDigiProducer.cc#L191

Reply:
The class PileupEvent has only getByLabel. Once it is upgraded to make use of token, the last reference to getByLabel can be moved out.